sys/include/riotboot/hdr.h

```diff
@@ -52,6 +57,9 @@ typedef struct {
     uint32_t magic_number;      /**< Header magic number (always "RIOT")              */
     uint32_t version;           /**< Integer representing the partition version       */
     uint32_t start_addr;        /**< Address after the allocated space for the header */
+    uint32_t chksum_legacy;     /**< Checksum of riotboot_hdr digest excluded         */
+    uint32_t img_size;          /**< Firmware size                                    */
```

Comment:
I was thinking about how we can avoid the `chksum_legacy` situation in the future, how about

```suggestion
    uint32_t img_size;          /**< Firmware size                                    */
    uint16_t header_len;        /**< Size of the riotboot header, excluding the checksum */
    uint16_t flags;
```

Reply:
What's the idea behind `flags`?

Reply:
There will be two padding bytes anyway, might as well put them to some use

Reply:
I'd use packed structs or call it `reserved` or `padding` with a comment why it's there. Or do we already have flags?

Reply:
If there is not use yet, I think `reserver` makes sense, changing the name later would be free. The idea would be to use the `header_len` field to known what to checksum?

Reply:
Something to consider though if the hdr_len field is modified, it could eventually calculate the checksum upon random memory regions

Reply:
The same goes for `img_size`. The implementation checking the image has to check if bounds aren't exceeded.